api/access/server.go

```diff
@@ -0,0 +1,515 @@
+		Timestamp:            timestamppb.New(header.Timestamp),
+		CollectionGuarantees: collections,
+		BlockSeals:           seals,
+		Signatures:           [][]byte{header.ParentVoterSig},
+	}
+
+	resp := access.BlockResponse{
+		Block: &block,
+	}
+
+	return &resp, nil
+}
+
+func (s *Server) GetCollectionByID(_ context.Context, in *access.GetCollectionByIDRequest) (*access.CollectionResponse, error) {
+	collId := flow.HashToID(in.Id)
+	collection, err := s.index.Collection(collId)
+	if err != nil {
+		return nil, fmt.Errorf("could not retrieve collection with ID %x: %w", in.Id, err)
+	}
+
+	collEntity := entities.Collection{
+		Id: in.Id,
+	}
+	for _, txID := range collection.Transactions {
+		collEntity.TransactionIds = append(collEntity.TransactionIds, conv.IDToHash(txID))
+	}
+
+	resp := access.CollectionResponse{
+		Collection: &collEntity,
+	}
+
+	return &resp, err
+}
+
+func (s *Server) GetTransaction(_ context.Context, in *access.GetTransactionRequest) (*access.TransactionResponse, error) {
+	txID := flow.HashToID(in.Id)
+	tx, err := s.index.Transaction(txID)
+	if err != nil {
+		return nil, fmt.Errorf("could not retrieve transaction: %w", err)
+	}
+
+	resp := access.TransactionResponse{
+		Transaction: convert.TransactionToMessage(*tx),
+	}
+
+	return &resp, nil
+}
+
+func (s *Server) GetTransactionResult(_ context.Context, in *access.GetTransactionRequest) (*access.TransactionResultResponse, error) {
+	txID := flow.HashToID(in.Id)
+	result, err := s.index.Result(txID)
+	if err != nil {
+		return nil, fmt.Errorf("could not retrieve transaction result: %w", err)
+	}
+
+	// We also need the height of the transaction we're looking at.
+	// TODO: See if it wouldn't make more sense to remove this index and just
+	//		 always return the status as sealed.
+	//		 https://github.com/optakt/flow-dps/issues/317
+	height, err := s.index.HeightForTransaction(txID)
+	if err != nil {
+		return nil, fmt.Errorf("could not retrieve block height: %w", err)
+	}
+
+	block, err := s.index.Header(height)
+	if err != nil {
+		return nil, fmt.Errorf("could not retrieve block header: %w", err)
+	}
+	blockID := block.ID()
+
+	statusCode := uint32(0)
+	if result.ErrorMessage == "" {
+		statusCode = 1
+	}
+
+	status := entities.TransactionStatus_SEALED
+	sealedHeight, err := s.index.Last()
+	if err != nil {
+		return nil, fmt.Errorf("could not retrieve last height: %w", err)
+	}
+
+	if height > sealedHeight {
+		status = entities.TransactionStatus_EXECUTED
+	}
+
+	events, err := s.index.Events(height)
+	if err != nil {
+		return nil, fmt.Errorf("could not retrieve events: %w", err)
+	}
+
+	resp := access.TransactionResultResponse{
+		Status:       status,
+		StatusCode:   statusCode,
+		ErrorMessage: result.ErrorMessage,
+		Events:       convert.EventsToMessages(events),
+		BlockId:      blockID[:],
+	}
+
+	return &resp, nil
+}
+
+func (s *Server) GetAccount(ctx context.Context, in *access.GetAccountRequest) (*access.GetAccountResponse, error) {
```

Comment:
I would change the signature to use the same types as the function we forward it to, and then change it to:

```go
func (s *Server) GetAccount(ctx context.Context, in *access.GetAccountAtLatestBlockRequest) (*access.AccountResponse, error) {
  return s.GetAccountAtLatestBlock(ctx, in)
}
```

GRPC should not be able to make the difference between two messages that have the exact same fields.